exporters/otlp/connection.go

```diff
@@ -79,35 +140,136 @@ func (e *Exporter) indefiniteBackgroundConnection() {
 		// 2. Otherwise block until we are disconnected, and
 		//    then retry connecting
 		select {
-		case <-e.stopCh:
+		case <-oc.stopCh:
 			return
 
-		case <-e.disconnectedCh:
+		default:
+			// Not stopped right now, go wait for
+			// disconnect signal then.
+		}
+
+		select {
+		case <-oc.stopCh:
+			return
+
+		case <-oc.disconnectedCh:
```

Comment:
This logic seems to check if stopped then enter a second select that checks if stopped again or disconnected. Is there a need for the first select statement's default case?

Reply:
First `select` is to quickly check if stopped. Without it, the second `select` could pick up the disconnected channel, even if stop channel is closed at the same time (which channel is picked up for reading first in `select` if both are readable is unspecified). That could result in going through the reconnecting routine unnecessarily.

Now I think I should actually move the first `select` after the second one, so if both channels are readable, we can do a quick check of the stopped channel in case disconnected channel is picked up for reading by `select`.

Reply:
> First `select` is to quickly check if stopped. Without it, the second `select` could pick up the disconnected channel, even if stop channel is closed at the same time (which channel is picked up for reading first in `select` if both are readable is unspecified). That could result in going through the reconnecting routine unnecessarily.

Right, makes sense :+1: 

> Now I think I should actually move the first `select` after the second one, so if both channels are readable, we can do a quick check of the stopped channel in case disconnected channel is picked up for reading by `select`.

:+1: